src/components/implementation/tests/pci/pci.c

```diff
@@ -0,0 +1,159 @@
+#include <cos_component.h>
+#include <llprint.h>
+#include <pci.h>
+
+struct pci_dev all_dev[6] = {
+		{
+			.bus = 0,	
+			.dev = 0,	
+			.func = 0,	
+			.vendor = 0x8086,	
+			.device = 0x1237,	
+			.classcode = 6,	
+			.subclass = 0,	
+			.progIF = 0,	
+			.header = 0,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		},
+		{
+			.bus = 0,	
+			.dev = 1,	
+			.func = 0,	
+			.vendor = 0x8086,	
+			.device = 0x7000,	
+			.classcode = 6,	
+			.subclass = 1,	
+			.progIF = 0,	
+			.header = 0x80,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		},
+		{
+			.bus = 0,	
+			.dev = 1,	
+			.func = 1,	
+			.vendor = 0x8086,	
+			.device = 0x7010,	
+			.classcode = 1,	
+			.subclass = 1,	
+			.progIF = 0x80,	
+			.header = 0,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		},
+		{
+			.bus = 0,	
+			.dev = 1,	
+			.func = 3,	
+			.vendor = 0x8086,	
+			.device = 0x7113,	
+			.classcode = 6,	
+			.subclass = 0x80,	
+			.progIF = 0,	
+			.header = 0,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		},
+		{
+			.bus = 0,	
+			.dev = 2,	
+			.func = 0,	
+			.vendor = 0x1234,	
+			.device = 0x1111,	
+			.classcode = 3,	
+			.subclass = 0,	
+			.progIF = 0,	
+			.header = 0,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		},
+		{
+			.bus = 0,	
+			.dev = 3,	
+			.func = 0,	
+			.vendor = 0x8086,	
+			.device = 0x100e,	
+			.classcode = 2,	
+			.subclass = 0,	
+			.progIF = 0,	
+			.header = 0,	
+			.data = {},	
+			.index = 0,	
+			.drvdata = 0
+		}
+	};
```

Comment:
It is a large enough structure, that having it in a separate file might make sense (as @WenyuanShao recommended above). Would give us an opportunity to choose a name to self-document slightly as well. `hw_profile.h` might be appropriate? The .h file would still be in this directory, FYI.